main/mio.c

```diff
@@ -999,8 +1004,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				s[i] = 0;
 				rv = s;
 			}
-			if (mio->impl.mem.pos >= mio->impl.mem.size)
+			if (!newline && pos >= buf_size)
```

Comment:
You can just simply set MAX_IN_MEMORY_FILE_SIZE to say 1 so getMio() always returns the file backend and you'll see different unit test results without this patch.